artworks/views.py

```diff
@@ -1,3 +1,64 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect, reverse
+from .forms import ArtForm
+from django.db import transaction
+from .models import Category, Artwork, ArtLike
+from django.http import HttpResponseRedirect
+from django.contrib.auth import authenticate
+
+
+def upload_view(request):
+    if request.method == "POST":
+        form = ArtForm(request.POST, request.FILES)
+        if form.is_valid():
+            with transaction.atomic():
+                form_save = form.save(commit=False)
+                tags = [tag.lower() for tag in form.cleaned_data['category'].replace(' ', '').split(',')]
+                form_save.uploader = request.user
+                form_save.save()
+                for tag in tags:
+                    category = Category.objects.filter(name=tag)
+                    if category.exists():
+                        form_save.category.add(category[0])
+                    else:
+                        category = Category.objects.create(name=tag)
+                        category.save()
+                        form_save.category.add(category)
+            return redirect('/')
+    else:
+        form = ArtForm()
+    context = {
+        "form":form,
+    }
+    return render(request, 'artworks/upload_art.html', context)
+
+
+def like_view(request, art_id):
+    if request.user.is_authenticated:
+        artwork = Artwork.objects.get(pk=art_id)
+        if artwork.likes.filter(id=request.user.id).exists():
+            artwork.likes.remove(request.user)
+        else:
+            artwork.likes.add(request.user)
```

Comment:
Adding `user = request.user` helps you a lot in not doing `request.user` all the time. It's also easier to read after.